cogs/syntax-tests.scm

```diff
@@ -0,0 +1,103 @@
+(require-builtin "steel/process")
+(require-builtin "steel/transducers")
+
+(require "tests/unit-test.scm"
+         (for-syntax "tests/unit-test.scm"))
+
+(define (eval-in-child datums)
```

Comment:
:information_source: I want to test expected failures when compiling macros. I don't think we have `eval` or something similar, so I've come up with a gimmick: just write the code to some file and run it in a separate process.

Reply:
So there is an `eval` - however it basically just works by embedding an engine instance itself, and then we act on that. More or less equivalent to what you've done, but runs in the same process: https://github.com/mattwparas/steel/blob/master/crates/steel-core/src/steel_vm/meta.rs#L225

You also can do something like this, where you pass in a list of expressions:


```
(run! (Engine::new) '((define (foo) 10) (foo))
```

I haven't tested it with macros and making sure all things line up, but it might be easier than creating another process.